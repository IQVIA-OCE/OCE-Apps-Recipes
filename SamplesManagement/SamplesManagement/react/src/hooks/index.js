export { useFetcher } from './useFetcher';
export { useBoolean } from './useBoolean';
export { useHandleData } from './useHandleData';
export { useBanner } from './useBanner';
