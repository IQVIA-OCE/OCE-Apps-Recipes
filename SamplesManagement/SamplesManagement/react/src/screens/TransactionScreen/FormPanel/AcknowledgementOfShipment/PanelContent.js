import React from 'react';
import { View, StyleSheet, Text } from 'react-native';
import { TextInput, Select } from 'apollo-react-native';
import { useFormikContext } from 'formik';
import DateField from '../DateField';
import { getFieldError, getFieldHelperText } from '../../utils';

const PanelContent = () => {
  const context = useFormikContext();
  const { values, handleChange, setFieldValue, errors, touched } = context;

  return (
    <View style={styles.container}>
      <View style={styles.col}>
        <DateField
          label="Received Date"
          value={values.fields.receivedDate}
          onChange={val => setFieldValue('fields.receivedDate', val)}
          style={styles.field}
          hasError={getFieldError('receivedDate', errors, touched)}
          helperText={getFieldHelperText('receivedDate', errors, touched)}
          required
          touched={touched}
        />
        <Select
          label="Condition of Package"
          placeholder={'-None-'}
          options={[
            { label: 'Undamaged', id: '1' },
            { label: 'Damaged', id: '2' },
            { label: 'Opened', id: '3' },
          ]}
          value={values.fields.conditionOfPackage}
          onChange={val => setFieldValue('fields.conditionOfPackage', val)}
          fullWidth
          style={{ width: '100%' }}
          error={getFieldError('conditionOfPackage', errors, touched)}
          helperText={getFieldHelperText('conditionOfPackage', errors, touched)}
          required
        />
      </View>
      <View
        style={{
          flex: 1,
        }}
      >
        <TextInput
          label="Comments"
          onChangeText={handleChange('fields.comments')}
          value={values.fields.comments}
          multiline
          fullWidth
        />
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    padding: 15,
    paddingHorizontal: 30,
    marginBottom: 10,
  },
  col: {
    borderRightWidth: 1,
    borderRightColor: 'rgb(221, 219, 218)',
    flex: 1,
    marginRight: 20,
    paddingRight: 20,
  },
  field: {
    marginBottom: 15,
  },
});

export default PanelContent;
