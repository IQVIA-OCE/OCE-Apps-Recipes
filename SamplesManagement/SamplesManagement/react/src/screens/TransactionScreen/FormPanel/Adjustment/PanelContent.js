import React from 'react';
import { View, StyleSheet } from 'react-native';
import { TextInput } from 'apollo-react-native';
import { useFormikContext } from 'formik';

const PanelContent = () => {
  const context = useFormikContext();
  const { values, setFieldValue } = context;

  return (
    <View style={styles.container}>
      <View style={styles.col}>
        <TextInput
          label="Comments"
          onChangeText={text => setFieldValue('fields.comments', text)}
          value={values.fields.comments}
          multiline
          fullWidth
        />
      </View>
      <View
        style={{
          flex: 1,
        }}
      ></View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    padding: 15,
    paddingHorizontal: 30,
    marginBottom: 10,
  },
  col: {
    flex: 1,
    marginRight: 20,
    paddingRight: 20,
  },
  field: {
    marginBottom: 15,
  },
});

export default PanelContent;
