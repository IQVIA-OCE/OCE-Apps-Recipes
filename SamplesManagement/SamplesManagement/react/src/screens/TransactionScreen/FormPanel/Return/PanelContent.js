import React from 'react';
import { View, StyleSheet } from 'react-native';
import { useFormikContext } from 'formik';
import { TextInput } from 'apollo-react-native';
import DateField from '../DateField';
import { getFieldError, getFieldHelperText } from '../../utils';

const PanelContent = ({ readonly }) => {
  const context = useFormikContext();
  const { values, setFieldValue, errors, touched } = context;

  return (
    <View style={styles.container} testID="ReturnPanelContent">
      <View style={styles.col}>
        <DateField
          label="Shipment Date"
          value={values.fields.shipmentDate}
          onChange={val => setFieldValue('fields.shipmentDate', val)}
          style={styles.field}
          hasError={getFieldError('shipmentDate', errors, touched)}
          helperText={getFieldHelperText('shipmentDate', errors, touched)}
          touched={touched}
          readonly={readonly}
        />
        <TextInput
          label="Tracking Number"
          onChangeText={val => setFieldValue('fields.trackingNumber', val)}
          value={values.fields.trackingNumber}
          readonly={readonly}
          fullWidth
        />
      </View>
      <View
        style={{
          flex: 1,
        }}
      >
        <TextInput
          label="Shipment Carrier"
          onChangeText={val => setFieldValue('fields.shipmentCarrier', val)}
          value={values.fields.shipmentCarrier}
          fullWidth
          style={styles.field}
          readonly={readonly}
        />
        <TextInput
          label="Comments"
          onChangeText={val => setFieldValue('fields.comments', val)}
          value={values.fields.comments}
          multiline
          fullWidth
          readonly={readonly}
        />
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    padding: 15,
    paddingHorizontal: 30,
    marginBottom: 10,
  },
  col: {
    borderRightWidth: 1,
    borderRightColor: 'rgb(221, 219, 218)',
    flex: 1,
    marginRight: 20,
    paddingRight: 20,
  },
  field: {
    marginBottom: 15,
  },
});

export default PanelContent;
