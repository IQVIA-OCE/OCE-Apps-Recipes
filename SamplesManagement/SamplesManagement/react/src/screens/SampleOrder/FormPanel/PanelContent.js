import React from 'react';
import { View, StyleSheet, Text } from 'react-native';
import { TextInput, Select, Autocomplete } from 'apollo-react-native';
import { useFormikContext } from 'formik';
import { useFetcher, useHandleData } from '../../../hooks';
import { fetchUserLocations } from '../../../api/SampleOrder';
import {
  normalizeUsers,
  normalizeLocations,
  getFieldError,
  getFieldHelperText,
} from '../utils';

const PanelContent = ({ readonly }) => {
  const context = useFormikContext();
  const { values, handleChange, setFieldValue, errors, touched } = context;

  const [locations] = useFetcher(async () => {
    return await fetchUserLocations(values.fields.user.Id);
  }, normalizeLocations);

  return (
    <View style={styles.container}>
      <View style={styles.col}>
        {useHandleData(locations)(data => {
          return readonly ? (
            <TextInput
              label="Ship To"
              value={values.fields.shipTo ? values.fields.shipTo.label : ''}
              fullWidth
              readonly
              style={styles.readonlyField}
            />
          ) : (
            <Select
              label="Ship To"
              placeholder={'-None-'}
              options={data}
              value={values.fields.shipTo}
              onChange={val => setFieldValue('fields.shipTo', val)}
              fullWidth
              style={{ width: '100%', marginBottom: 15 }}
              error={getFieldError('shipTo', errors, touched)}
              helperText={getFieldHelperText('shipTo', errors, touched)}
              required
            />
          );
        })}
      </View>
      <View
        style={{
          flex: 1,
        }}
      >
        <TextInput
          label="Comments"
          onChangeText={handleChange('fields.comments')}
          value={values.fields.comments}
          multiline
          fullWidth
          readonly={readonly}
          style={readonly ? styles.readonlyWithBorder : styles.field}
        />
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    padding: 15,
    paddingHorizontal: 30,
    marginBottom: 10,
  },
  col: {
    borderRightWidth: 1,
    borderRightColor: 'rgb(221, 219, 218)',
    flex: 1,
    marginRight: 20,
    paddingRight: 20,
  },
  field: {
    marginBottom: 15,
  },
  readonlyField: {
    marginBottom: 18,
  },
  readonlyWithBorder: {
    marginBottom: 10,
    borderBottomWidth: 0.5,
    borderColor: '#D9D9D9',
  },
});

export default PanelContent;
