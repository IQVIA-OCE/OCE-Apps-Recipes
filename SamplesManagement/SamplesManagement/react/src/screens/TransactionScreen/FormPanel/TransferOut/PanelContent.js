import React, { useEffect } from 'react';
import { View, StyleSheet, Text } from 'react-native';
import { TextInput, Select, Autocomplete } from 'apollo-react-native';
import { useFormikContext } from 'formik';
import DateField from '../DateField';
import { useFetcher, useHandleData } from '../../../../hooks';
import {
  fetchAllUsers,
  fetchUserLocations,
  fetchUserTerritory,
} from '../../../../api/SampleTransaction';
import {
  normalizeUsers,
  normalizeLocations,
  getFieldError,
  getFieldHelperText,
} from '../../utils';

const PanelContent = ({ readonly }) => {
  const {
    values,
    handleChange,
    setFieldValue,
    errors,
    touched,
  } = useFormikContext();

  const [users] = useFetcher(fetchAllUsers, normalizeUsers);

  const toSalesRepId = values.fields.toSalesRep
    ? values.fields.toSalesRep['value']
    : null;

  const [locations, locationsActions] = useFetcher(
    async () => await fetchUserLocations(toSalesRepId),
    normalizeLocations
  );
  const [territory, territoryActions] = useFetcher(
    async () => await fetchUserTerritory(toSalesRepId)
  );

  useEffect(() => {
    locationsActions.handleFetch();
    territoryActions.handleFetch();
  }, [values.fields.toSalesRep]);

  useEffect(() => {
    const toSalesRepTerritory =
      territory.data && territory.data.length
        ? territory.data[0].OCE__Territory__c
        : '';
    setFieldValue('fields.toSalesRepTerritory.name', toSalesRepTerritory);
  }, [territory]);

  return (
    <View style={styles.container}>
      <View style={styles.col}>
        {useHandleData(users)(data => {
          return readonly ? (
            <TextInput
              label="To Sales Rep"
              value={
                values.fields.toSalesRep ? values.fields.toSalesRep.label : ''
              }
              fullWidth
              readonly
              style={styles.readonlyField}
            />
          ) : (
            <Autocomplete
              label="To Sales Rep"
              helperText={getFieldHelperText('toSalesRep', errors, touched)}
              error={getFieldError('toSalesRep', errors, touched)}
              placeholder=""
              source={data}
              onChange={val => {
                setFieldValue('fields.shipTo', null);
                setFieldValue('fields.toSalesRep', val);
              }}
              singleSelect
              singleSelectValue={values.fields.toSalesRep}
              fullWidth
              required
              style={styles.field}
            />
          );
        })}
        {useHandleData(locations)(data => {
          return readonly ? (
            <TextInput
              label="Ship To"
              value={values.fields.shipTo ? values.fields.shipTo.label : ''}
              fullWidth
              readonly
              style={styles.readonlyField}
            />
          ) : (
            <Select
              label="Ship To"
              placeholder={'-None-'}
              options={data}
              value={values.fields.shipTo}
              onChange={val => setFieldValue('fields.shipTo', val)}
              fullWidth
              style={{ width: '100%', marginBottom: 15 }}
              error={getFieldError('shipTo', errors, touched)}
              helperText={getFieldHelperText('shipTo', errors, touched)}
              disabled={values.fields.toSalesRep == null || users.loading}
              required
            />
          );
        })}

        <DateField
          label="Shipment Date"
          value={values.fields.shipmentDate}
          onChange={val => setFieldValue('fields.shipmentDate', val)}
          style={styles.field}
          hasError={getFieldError('shipmentDate', errors, touched)}
          helperText={getFieldHelperText('shipmentDate', errors, touched)}
          touched={touched}
          readonly={readonly}
          required
        />

        <TextInput
          label="Shipment Carrier"
          onChangeText={handleChange('fields.shipmentCarrier')}
          value={values.fields.shipmentCarrier}
          fullWidth
          error={getFieldError('shipmentCarrier', errors, touched)}
          helperText={getFieldHelperText('shipmentCarrier', errors, touched)}
          required
          readonly={readonly}
          style={readonly ? styles.readonlyField : styles.field}
        />
      </View>
      <View
        style={{
          flex: 1,
        }}
      >
        {useHandleData(territory)(() => {
          return (
            <TextInput
              label="To Sales Rep Territory"
              value={values.fields.toSalesRepTerritory.name}
              fullWidth
              style={[styles.field, { borderWidth: 0, opacity: 1 }]}
              readonly
              editable={false}
            />
          );
        })}

        <TextInput
          label="Tracking Number"
          onChangeText={handleChange('fields.trackingNumber')}
          value={values.fields.trackingNumber}
          error={getFieldError('trackingNumber', errors, touched)}
          helperText={getFieldHelperText('trackingNumber', errors, touched)}
          fullWidth
          required
          readonly={readonly}
          style={readonly ? styles.readonlyField : styles.field}
        />
        <TextInput
          label="Comments"
          onChangeText={handleChange('fields.comments')}
          value={values.fields.comments}
          multiline
          fullWidth
          readonly={readonly}
          style={readonly ? styles.readonlyWithBorder : styles.field}
        />
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    padding: 15,
    paddingHorizontal: 30,
    marginBottom: 10,
  },
  col: {
    borderRightWidth: 1,
    borderRightColor: 'rgb(221, 219, 218)',
    flex: 1,
    marginRight: 20,
    paddingRight: 20,
  },
  field: {
    marginBottom: 15,
  },
  readonlyField: {
    marginBottom: 18,
  },
  readonlyWithBorder: {
    marginBottom: 10,
    borderBottomWidth: 0.5,
    borderColor: '#D9D9D9',
  },
});

export default PanelContent;
