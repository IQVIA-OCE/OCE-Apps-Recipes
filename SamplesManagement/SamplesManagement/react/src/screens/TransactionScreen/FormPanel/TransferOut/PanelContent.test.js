import React from 'react';
import PanelContent from './PanelContent';
import { Formik } from 'formik';
import { useBoolean } from '../../../../hooks';
import { Autocomplete, Select } from '@oce-apps/apollo-react-native';
import { render, act, fireEvent } from '@testing-library/react-native';
import moment from 'moment';

jest.mock('../../../../hooks/useBoolean');

describe('PanelHeader', () => {
  beforeAll(() => {
    useBoolean.mockReturnValue([
      false,
      { setTrue: jest.fn(), setFalse: jest.fn() },
    ]);
  });

  it('should render properly', () => {
    const Component = () => (
      <Formik
        enableReinitialize
        initialValues={{
          fields: {
            fromSalesRep: '',
            shipTo: '',
            receivedDate: moment().toISOString(),
            conditionOfPackage: '',
            comments: '',
            toSalesRepTerritory: {
              name: "Atalanta"
            },
            transactionRepTerritory: {
              name: "TerrritoryName"
            },
            fromSalesRepTerritory: {
              name: "Atalanta"
            }
          },
        }}
      >
        <PanelContent />
      </Formik>
    );

    const { UNSAFE_root, getByText } = render(
      <Component />
    );
    act(() => jest.runAllTimers());

    fireEvent(UNSAFE_root.findByType(Autocomplete), 'change');
    act(() => jest.runAllTimers());

    fireEvent(UNSAFE_root.findAllByType(Select)[0], 'change');
    act(() => jest.runAllTimers());

    fireEvent(UNSAFE_root.findAllByType(Select)[0], 'change');
    act(() => jest.runAllTimers());

    expect(getByText(/Shipment Carrier/)).toBeTruthy();
  });
});
