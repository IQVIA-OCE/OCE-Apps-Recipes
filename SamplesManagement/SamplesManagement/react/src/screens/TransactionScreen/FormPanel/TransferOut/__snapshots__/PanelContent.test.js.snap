// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PanelHeader should render properly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "marginBottom": 10,
      "padding": 15,
      "paddingHorizontal": 30,
    }
  }
>
  <View
    style={
      Object {
        "borderRightColor": "rgb(221, 219, 218)",
        "borderRightWidth": 1,
        "flex": 1,
        "marginRight": 20,
        "paddingRight": 20,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "marginBottom": 15,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "flexWrap": "wrap",
          }
        }
      />
      <View
        collapsable={false}
      >
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "width": "100%",
              },
              Object {
                "zIndex": 1,
              },
              Object {
                "opacity": 1,
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "marginBottom": 8,
              }
            }
          >
            To Sales Rep
            <Text
              style={
                Object {
                  "color": "#E20000",
                }
              }
            >
               *
            </Text>
          </Text>
          <View
            style={
              Array [
                Object {
                  "borderColor": "#D9D9D9",
                  "borderRadius": 4,
                  "borderWidth": 1,
                },
                Object {
                  "zIndex": 1,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "height": 40,
                  "minHeight": 40,
                  "paddingBottom": 0,
                  "paddingTop": 0,
                }
              }
            >
              <TextInput
                allowFontScaling={true}
                defaultValues={Array []}
                editable={true}
                keyboardType="default"
                maxResizableWidth={750}
                minHeight={60}
                minWidth={176}
                onBlur={[Function]}
                onChange={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                placeholder=""
                placeholderTextColor="rgba(0, 0, 0, 0.54)"
                rejectResponderTermination={true}
                selectionColor="#0768FD"
                singleSelect={true}
                source={Array []}
                style={
                  Array [
                    Object {
                      "elevation": 5,
                      "margin": 0,
                      "paddingLeft": 8,
                      "shadowColor": "#000",
                      "shadowOffset": Object {
                        "height": 2,
                        "width": 0,
                      },
                      "shadowOpacity": 0.25,
                      "shadowRadius": 3.84,
                      "textAlign": "left",
                      "zIndex": 5,
                    },
                    Object {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "height": "95%",
                      "paddingRight": 0,
                      "textAlignVertical": "center",
                      "width": "100%",
                    },
                  ]
                }
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "width": "100%",
          },
          Object {
            "marginBottom": 15,
            "width": "100%",
          },
        ]
      }
    >
      <View>
        <Text
          style={
            Array [
              Object {
                "marginBottom": 8,
              },
              Object {
                "color": "rgba(0, 0, 0, 0.4)",
                "fontFamily": "System",
                "fontWeight": "400",
              },
            ]
          }
        >
          Ship To
          <Text
            style={
              Object {
                "color": "#E20000",
              }
            }
          >
             *
          </Text>
        </Text>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              },
              Object {
                "backgroundColor": "rgba(246, 246, 246, 0.4)",
                "borderColor": "rgba(217, 217, 217, 0.4)",
                "borderRadius": 4,
                "borderWidth": 1,
                "elevation": 0,
                "height": 40,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "stretch",
                "flex": 1,
                "flexDirection": "row",
                "flexGrow": 1,
                "paddingLeft": 8,
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              hitSlop={
                Object {
                  "bottom": 14,
                  "left": 14,
                  "right": 14,
                  "top": 14,
                }
              }
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "flex": 1,
                  "opacity": 1,
                }
              }
            >
              <Text
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "color": "rgba(0, 0, 0, 0.4)",
                      "fontFamily": "System",
                      "fontSize": 18,
                      "fontWeight": "400",
                      "paddingLeft": 8,
                    },
                  ]
                }
              >
                -None-
              </Text>
            </View>
          </View>
          <View
            accessibilityComponentType="button"
            accessibilityRole="button"
            accessibilityStates={
              Array [
                "disabled",
              ]
            }
            accessibilityTraits="button"
            accessible={true}
            centered={true}
            focusable={true}
            hitSlop={
              Object {
                "bottom": 6,
                "left": 6,
                "right": 6,
                "top": 6,
              }
            }
            onClick={[Function]}
            onLayout={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            pointerEvents="box-only"
          >
            <View
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Array [
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                      "margin": 6,
                      "overflow": "hidden",
                    },
                    Object {
                      "borderRadius": 18,
                      "height": 36,
                      "width": 36,
                    },
                    Object {
                      "opacity": 0.32,
                    },
                    Object {
                      "margin": 0,
                    },
                  ],
                ]
              }
            >
              <View>
                <Text
                  accessibilityElementsHidden={true}
                  allowFontScaling={false}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  style={
                    Array [
                      Object {
                        "color": "rgba(0, 0, 0, 0.4)",
                        "fontSize": 24,
                      },
                      Array [
                        Object {
                          "transform": Array [
                            Object {
                              "scaleX": 1,
                            },
                          ],
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                      ],
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  ï…€
                </Text>
              </View>
            </View>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "bottom": 0,
                    "left": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "borderRadius": 0,
                  },
                ]
              }
            />
          </View>
        </View>
      </View>
      <View
        collapsable={false}
      />
    </View>
    <View
      style={
        Array [
          Object {},
          Object {
            "marginBottom": 15,
          },
        ]
      }
    >
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "width": "100%",
            },
            undefined,
            Object {
              "opacity": 1,
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "marginBottom": 8,
            }
          }
        >
          Shipment Date
          <Text
            style={
              Object {
                "color": "#E20000",
              }
            }
          >
             *
          </Text>
        </Text>
        <View
          style={
            Array [
              Object {
                "borderColor": "#D9D9D9",
                "borderRadius": 4,
                "borderWidth": 1,
              },
              Object {},
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 40,
                "minHeight": 40,
                "paddingBottom": 0,
                "paddingTop": 0,
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              editable={true}
              maxResizableWidth={750}
              minHeight={60}
              minWidth={176}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholderTextColor="rgba(0, 0, 0, 0.54)"
              rejectResponderTermination={true}
              selectionColor="#0768FD"
              style={
                Array [
                  Object {
                    "elevation": 5,
                    "margin": 0,
                    "paddingLeft": 8,
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 2,
                      "width": 0,
                    },
                    "shadowOpacity": 0.25,
                    "shadowRadius": 3.84,
                    "textAlign": "left",
                    "zIndex": 5,
                  },
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "height": "95%",
                    "paddingRight": 40,
                    "textAlignVertical": "center",
                    "width": "100%",
                  },
                ]
              }
              underlineColorAndroid="transparent"
              value=""
            />
            <View
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                  "position": "absolute",
                  "right": 10,
                  "top": "50%",
                  "transform": Array [
                    Object {
                      "translateY": -12.5,
                    },
                  ],
                  "zIndex": 5,
                }
              }
            >
              <Text
                accessibilityElementsHidden={true}
                allowFontScaling={false}
                importantForAccessibility="no-hide-descendants"
                pointerEvents="none"
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontSize": 25,
                    },
                    Array [
                      Object {
                        "transform": Array [
                          Object {
                            "scaleX": 1,
                          },
                        ],
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                    ],
                    Object {
                      "fontFamily": "Material Design Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                ïƒ®
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View>
        <View />
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "width": "100%",
          },
          Object {
            "marginBottom": 15,
          },
          Object {
            "opacity": 1,
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "marginBottom": 8,
          }
        }
      >
        Shipment Carrier
        <Text
          style={
            Object {
              "color": "#E20000",
            }
          }
        >
           *
        </Text>
      </Text>
      <View
        style={
          Array [
            Object {
              "borderColor": "#D9D9D9",
              "borderRadius": 4,
              "borderWidth": 1,
            },
            Object {},
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 40,
              "minHeight": 40,
              "paddingBottom": 0,
              "paddingTop": 0,
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            editable={true}
            maxResizableWidth={750}
            minHeight={60}
            minWidth={176}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            placeholderTextColor="rgba(0, 0, 0, 0.54)"
            rejectResponderTermination={true}
            selectionColor="#0768FD"
            style={
              Array [
                Object {
                  "elevation": 5,
                  "margin": 0,
                  "paddingLeft": 8,
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 0,
                  },
                  "shadowOpacity": 0.25,
                  "shadowRadius": 3.84,
                  "textAlign": "left",
                  "zIndex": 5,
                },
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "height": "95%",
                  "paddingRight": 0,
                  "textAlignVertical": "center",
                  "width": "100%",
                },
              ]
            }
            underlineColorAndroid="transparent"
          />
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "width": "100%",
          },
          Array [
            Object {
              "marginBottom": 15,
            },
            Object {
              "borderWidth": 0,
              "opacity": 1,
            },
          ],
          Object {
            "opacity": 1,
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "marginBottom": 8,
          }
        }
      >
        To Sales Rep Territory
      </Text>
      <View
        style={
          Array [
            Object {
              "borderColor": "#D9D9D9",
              "borderRadius": 4,
              "borderWidth": 1,
            },
            Object {
              "borderWidth": 0,
              "opacity": 1,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 40,
              "minHeight": 40,
              "paddingBottom": 0,
              "paddingTop": 0,
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            editable={false}
            maxResizableWidth={750}
            minHeight={60}
            minWidth={176}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            placeholderTextColor="rgba(0, 0, 0, 0.54)"
            readonly={true}
            rejectResponderTermination={true}
            selectionColor="#0768FD"
            style={
              Array [
                Object {
                  "elevation": 5,
                  "margin": 0,
                  "paddingLeft": 8,
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 0,
                  },
                  "shadowOpacity": 0.25,
                  "shadowRadius": 3.84,
                  "textAlign": "left",
                  "zIndex": 5,
                },
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "height": "95%",
                  "paddingRight": 0,
                  "textAlignVertical": "center",
                  "width": "100%",
                },
              ]
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "width": "100%",
          },
          Object {
            "marginBottom": 15,
          },
          Object {
            "opacity": 1,
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "marginBottom": 8,
          }
        }
      >
        Tracking Number
        <Text
          style={
            Object {
              "color": "#E20000",
            }
          }
        >
           *
        </Text>
      </Text>
      <View
        style={
          Array [
            Object {
              "borderColor": "#D9D9D9",
              "borderRadius": 4,
              "borderWidth": 1,
            },
            Object {},
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 40,
              "minHeight": 40,
              "paddingBottom": 0,
              "paddingTop": 0,
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            editable={true}
            maxResizableWidth={750}
            minHeight={60}
            minWidth={176}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            placeholderTextColor="rgba(0, 0, 0, 0.54)"
            rejectResponderTermination={true}
            selectionColor="#0768FD"
            style={
              Array [
                Object {
                  "elevation": 5,
                  "margin": 0,
                  "paddingLeft": 8,
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 0,
                  },
                  "shadowOpacity": 0.25,
                  "shadowRadius": 3.84,
                  "textAlign": "left",
                  "zIndex": 5,
                },
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "height": "95%",
                  "paddingRight": 0,
                  "textAlignVertical": "center",
                  "width": "100%",
                },
              ]
            }
            underlineColorAndroid="transparent"
          />
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "width": "100%",
          },
          undefined,
          Object {
            "opacity": 1,
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "marginBottom": 8,
          }
        }
      >
        Comments
      </Text>
      <View
        style={
          Array [
            Object {
              "borderColor": "#D9D9D9",
              "borderRadius": 4,
              "borderWidth": 1,
            },
            Object {},
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 80,
              "minHeight": 80,
              "paddingBottom": 0,
              "paddingTop": 0,
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            editable={true}
            maxResizableWidth={750}
            minHeight={60}
            minWidth={176}
            multiline={true}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            placeholderTextColor="rgba(0, 0, 0, 0.54)"
            rejectResponderTermination={true}
            selectionColor="#0768FD"
            style={
              Array [
                Object {
                  "elevation": 5,
                  "margin": 0,
                  "paddingLeft": 8,
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 0,
                  },
                  "shadowOpacity": 0.25,
                  "shadowRadius": 3.84,
                  "textAlign": "left",
                  "zIndex": 5,
                },
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "height": "95%",
                  "paddingRight": 0,
                  "textAlignVertical": "center",
                  "width": "100%",
                },
              ]
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;
